include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -1557,46 +1647,142 @@ FEPointEvaluation<n_components_, dim, spacedim, Number>::evaluate_fast(
   const EvaluationFlags::EvaluationFlags &evaluation_flags)
 {
   // fast path with tensor product evaluation
-  if (solution_renumbered.size() != dofs_per_component)
-    solution_renumbered.resize(dofs_per_component);
+  if (use_face_path)
+    {
+      if (solution_renumbered.size() != 3 * dofs_per_component_face)
+        solution_renumbered.resize(3 * dofs_per_component_face);
+    }
+  else
+    {
+      if (solution_renumbered.size() != dofs_per_component)
+        solution_renumbered.resize(dofs_per_component);
+    }
   for (unsigned int comp = 0; comp < n_components; ++comp)
     {
       const unsigned int *renumber_ptr =
         renumber.data() +
         (component_in_base_element + comp) * dofs_per_component;
-      for (unsigned int i = 0; i < dofs_per_component; ++i)
-        ETT::read_value(solution_values[renumber_ptr[i]],
-                        comp,
-                        solution_renumbered[i]);
+      if (use_face_path)
+        {
+          ScalarNumber *input  = scratch_data_scalar.begin();
+          ScalarNumber *output = input + dofs_per_component;
+
+          for (unsigned int i = 0; i < dofs_per_component; ++i)
+            input[i] = solution_values[renumber_ptr[i]];
+
+          internal::FEFaceNormalEvaluationImpl<dim, -1, ScalarNumber>::
+            template interpolate<true, false>(1,
+                                              evaluation_flags,
+                                              shape_info,
+                                              input,
+                                              output,
+                                              current_face_number);
+
+          for (unsigned int i = 0; i < 3 * dofs_per_component_face; ++i)
+            ETT::read_value(output[i], comp, solution_renumbered[i]);
```

Comment:
Of course it should be possible. I would consider this step a optimization which is more involved because it also affects the `internal::evaluate_tensor_product_value_and_gradient_...` functions.